google/cloud/bigquery/client.py

```diff
@@ -596,22 +612,33 @@ def create_table(self, table, exists_ok=False, retry=DEFAULT_RETRY, timeout=None
         data = table.to_api_repr()
         try:
             span_attributes = {"path": path, "dataset_id": dataset_id}
-            with create_span(
-                name="BigQuery.createTable", attributes=span_attributes, client=self
-            ):
-                api_response = self._call_api(
-                    retry, method="POST", path=path, data=data, timeout=timeout
-                )
+            api_response = self._call_api(
+                retry,
+                span_name="BigQuery.createTable",
+                span_attributes=span_attributes,
+                method="POST",
+                path=path,
+                data=data,
+                timeout=timeout,
+            )
             return Table.from_api_repr(api_response)
         except google.api_core.exceptions.Conflict:
             if not exists_ok:
                 raise
             return self.get_table(table.reference, retry=retry)
 
-    def _call_api(self, retry, **kwargs):
+    def _call_api(
+        self, retry, span_name=None, span_attributes=None, job_ref=None, **kwargs
+    ):
+
         call = functools.partial(self._connection.api_request, **kwargs)
         if retry:
             call = retry(call)
+        if span_name is not None:
+            with create_span(
+                name=span_name, attributes=span_attributes, client=self, job_ref=job_ref
+            ):
+                return call()
         return call()
```

Comment:
Could you add a unit test just for `_call_api` itself to check this code branch? It can verify that `create_span` is not called if `span_name` is not specified.

Reply:
test's added in `test_client.py` seen on line 253 -> 299